src/main/java/com/spotinst/sdkjava/example/ElastigroupUsageExampleAzure.java

```diff
@@ -348,6 +414,158 @@ private static void sleep(Integer seconds) {
+
+        return vmHeathResponse;
+    }
+
+    public static void suspendGroup(SpotinstElastigroupClientAzure elastigroupClient, String groupId) {
+        SuspendGroupProcessesAzure.Builder suspendgroupBuilder = SuspendGroupProcessesAzure.Builder.get();
+        SuspendGroupProcessesAzure suspendProcessRequest = suspendgroupBuilder.setName(ProcessNameEnumsAzure.autoHealing)
+                .setTtlInMinutes(120)
+                .build();
+        List<SuspendGroupProcessesAzure> suspendProcess = new ArrayList<>();
+        suspendProcess.add(suspendProcessRequest);
+        SuspendgroupRequestAzure.Builder requestBuilder = SuspendgroupRequestAzure.Builder.get();
+        SuspendgroupRequestAzure suspendGroupRequest = requestBuilder.setSuspendGroup(suspendProcess)
+                .setGroupId(groupId).build();
+
+        Boolean suspended = elastigroupClient.suspendGroup(suspendGroupRequest);
+        if (suspended) {
+            System.out.println("Group Process suspended Successfully.");
+        }
+    }
+
+    public static void resumeGroup(SpotinstElastigroupClientAzure elastigroupClient, String groupId) {
+        ResumeGroupProcessesAzure.Builder resumegroupBuilder = ResumeGroupProcessesAzure.Builder.get();
+        ResumeGroupProcessesAzure resumeProcessRequest = resumegroupBuilder.setName(ProcessNameEnumsAzure.autoHealing)
+                .build();
+        List<ResumeGroupProcessesAzure> resumeProcess = new ArrayList<>();
+        resumeProcess.add(resumeProcessRequest);
+        ResumegroupRequestAzure.Builder requestBuilder = ResumegroupRequestAzure.Builder.get();
+        ResumegroupRequestAzure resumeGroupRequest = requestBuilder.setResumeGroup(resumeProcess)
+                .setGroupId(groupId).build();
+
+        Boolean resumed = elastigroupClient.resumeGroup(resumeGroupRequest );
+        if (resumed) {
+            System.out.println("Group Process resumed Successfully.");
+        }
+    }
+
+    public static void vmProtection(SpotinstElastigroupClientAzure elastigroupClient, String groupId) {
+
+        Boolean vmProtected = elastigroupClient.vmProtection(groupId, "vm-51e1b5e61973", 180);
+        if (vmProtected) {
+            System.out.println("Virtual Machine is Protected Successfully.");
+        }
+    }
+
+    public static void removeVmProtection(SpotinstElastigroupClientAzure elastigroupClient, String groupId, String vmName) {
+
+        Boolean vmProtectedRemoved = elastigroupClient.vmRemoveProtection(groupId, vmName);
+        if (vmProtectedRemoved) {
+            System.out.println("Virtual Machine is Protection Removed Successfully.");
+        }
+    }
+
+    public static ElastigroupAzure importFromVm(SpotinstElastigroupClientAzure elastigroupClient) {
+        ElastigroupAzure elastigroupImportResponse = elastigroupClient
+                .importGroupFromVirtalMachine("AutomationResourceGroup", "vm-17793a03a276");
+
```

Comment:
why vm name here static

Reply:
This API has issue in product, testing is pending so it is hardcoded for now.